internals/secrethub/tpl/v2.go

```diff
@@ -0,0 +1,113 @@
+package tpl
+
+import (
+	"fmt"
+	"github.com/secrethub/secrethub-cli/internals/tpl"
+)
+
+// NewV2Parser returns a parser for the v2 template syntax.
+//
+// V2 templates can contain secret paths between brackets:
+// {{ path/to/secret }}
+//
+// Secret paths can contain variables between ${}:
+// ${ var }
+//
+// Combined this can look like:
+// {{ ${app}/db/secret }}
```

Comment:
Does the `v2` syntax support variables outside of brackets? E.g. something like this:

```
DB_USER = ${env}-user
DB_PASSWORD = {{ company/repo/${env}/app/db_password }}
```

When we talk about writing a spec for the syntax, this is one of the things we mean. Consider expanding this GoDoc comment to a bit more complete description of the syntax and what behavior people can expect (e.g. are spaces ignored in vars `${ foo }` and in path substitutions `{{ foo }}`. 

Then we can later create more complete user documentation from the code documentation.

Reply:
> Does the v2 syntax support variables outside of brackets?

It does not. I've updated the godoc to explicitly say this.

Spaces directly after opening delimiters and directly before closing delimiters are ignored. I've added a comment about this to the godoc.

Reply:
> > Does the v2 syntax support variables outside of brackets?  

> It does not. I've updated the godoc to explicitly say this.

Isn't this something we want? Can't we easily solve this by swapping the two template parsers around? So first variable substitution and _then_ secrets?

Reply:
Yes that makes absolute sense to me. First do variable substition and then do secret injection.